### PR TITLE
Fix create() method for Bundle type transaction/batch

### LINKED ISSUE #102: The create() method does not work for Bundle transactions
This has already been noted in #54. Calling the create() method on a Bundle with type "transaction" results in a POST to /fhir/Bundle instead of /fhir (spec: http://hl7.org/fhir/http.html#transaction)

HAPI FHIR has this error  "Unable to store a Bundle resource on this server with a Bundle.type value of: transaction"

I'm happy to submit a new PR. #54 seems dead.

### PR DESCRIPTION
**Pull request by martinburchell**
Supersedes #54, fixes #102

I'm not sure this is much of an improvement on #54 . I've added a createUrl() method as @p2 suggested. This might not have been what was intended. To me it would make sense for createUrl() to return relativeBase() in fhirabstractresource.py and for the Bundle class to override this. To do that I'd need some help working that into the resource template / generate_models script.

I would also add some tests but I need some help working out how to run the tests.

### CODE REVIEW

**Comment by KeironO** [comment]
Not sure if this is the right place to ask, so if @ducu believes this is misdirected feel free to remove it.

I work for Cwm Taf Morgannwg Health Board and we're currently at the start of our FHIR journey. We have been using this library to build FHIR capabilities from internally developed software packages with a great deal with success. However, one of the major stumbling blocks is the inability to submit FHIR Bundles to our SMART-on-FHIR enabled server.

I've only done a small amount of testing, but this PR successfully enabled us to upload some test bundles to our testing server.

Whilst we can just continue working from this branch, it would be ideal if it was merged into the master branch ready for the next version to be pushed so when we update our packages it'll enable us to start exploiting Bundles to their fullest. I have no problem helping getting this merged, so if more work is required we'd love to help.

Thanks.

Keiron

**Comment by ducu** [comment]
Anyone here can please remove whitespace changes from this PR?

**Comment by martinburchell** [comment]
> Anyone here can please remove whitespace changes from this PR? Maybe @KeironO?

I'll do it.

File: fhir-parser-resources/fhirabstractresource.py [hunk 1]
```diff
@@ -110,7 +110,19 @@ def read_from(cls, path, server):
         instance = cls(jsondict=ret)
         instance.origin_server = server
         return instance
-    
+
+    def createUrl(self):
```
Comment (ducu) [inline fhir-parser-resources/fhirabstractresource.py:114]: Maybe a better name here is `createPath`? The return is not a url but a path used in urljoin later
https://github.com/smart-on-fhir/client-py/blob/b4da2bdc3bf024b6d0a78c85d80bf513302c1b13/fhirclient/server.py#L228

**Review by ducu — APPROVED** [review]

**Review by ducu — APPROVED** [review]
Ok will merge this, thanks!

**Pull request merged**